[DOC] Add Streaming doc for Tempo data source (#91432)
grafana/grafana: docs/sources/shared/datasources/tempo-search-traceql.md

The Tempo data source supports streaming responses to TraceQL queries so you can see partial query results as they come in without waiting for the whole query to finish.

{{% admonition type="note" %}}

> To use this experimental feature, enable the `traceQLStreaming` feature toggle. When active, all configured Tempo data sources will attempt to use streaming.

Reviewer: ```suggestion
To use this experimental feature, enable the `traceQLStreaming` feature toggle. When active, all configured Tempo data sources will attempt to use streaming. You can control which Tempo datasources do and don't attempt to stream results at the per-datasource level using the **Streaming** section of the Tempo datasource configuration. 
```

> In Grafana Cloud, you can enable this capability in the **Streaming** section of the Tempo data source settings.

Reviewer: ```suggestion
```

{{% /admonition %}}

> Streaming is available for both the **Search** and **TraceQL** query types.

Reviewer: ```suggestion
Results will be streamed when using both the **Search** and **TraceQL** query interfaces in Explore.
```

Reviewer: I don't really think of them as different "types" so much as different ways to run queries. One is more like a WIZYWIG way to build queries / graphical query builder, and the other is more like the command-line where you type the query into a blank text box.

Author: The UI calls them query types. 
![image](https://github.com/user-attachments/assets/630c2e68-68c4-4626-bb29-8bb73836f806)

Reviewer: Ah good point. Resolving then.
